yt_dlp/extractor/viously.py

```diff
@@ -0,0 +1,38 @@
+from .common import InfoExtractor
+from ..utils import (
+    get_element_html_by_class,
+    get_elements_html_by_class,
+)
+
+
+class ViouslyIE(InfoExtractor):
+    _VALID_URL = False
+    _API_URL = 'https://www.viously.com/video/hls/{0:}/index.m3u8'
+    _WEBPAGE_TESTS = [{
+        'url': 'http://www.turbo.fr/videos-voiture/454443-turbo-du-07-09-2014-renault-twingo-3-bentley-continental-gt-speed-ces-guide-achat-dacia.html',
+        'md5': '37a6c3381599381ff53a7e1e0575c0bc',
+        'info_dict': {
+            'id': 'F_xQzS2jwb3',
+            'ext': 'mp4',
+            'title': 'Turbo du 07/09/2014 : Renault Twingo 3, Bentley Continental GT Speed, CES, Guide Achat Dacia...',
+            'description': 'Turbo du 07/09/2014 : Renault Twingo 3, Bentley Continental GT Speed, CES, Guide Achat Dacia...',
+            'age_limit': 0,
+            'upload_date': str,
+            'timestamp': float,
+        }
+    }]
+
+    def _extract_from_webpage(self, url, webpage):
+        has_vously_player = get_element_html_by_class('viously-player', webpage) or get_element_html_by_class('vsly-player', webpage)
+        if not has_vously_player:
+            return
+        viously_players = get_elements_html_by_class('viously-player', webpage) + get_elements_html_by_class('vsly-player', webpage)
+        for viously_player in viously_players:
+            video_id = self._html_search_regex(r'id="([-_\w]+)"', viously_player, 'video_id')
+            title = self._html_extract_title(webpage)
+            yield {
+                'id': video_id,
+                'title': title,
+                'description': title,
+                'formats': self._extract_m3u8_formats(self._API_URL.format(video_id), video_id),
+            }
```

Comment:
Here I would actually call their API to extract the metadata
```suggestion
        def custom_decode(text):
            STANDARD_ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/='
            CUSTOM_ALPHABET = 'VIOUSLYABCDEFGHJKMNPQRTWXZviouslyabcdefghjkmnpqrtwxz9876543210+/='
            data = base64.b64decode(text.translate(str.maketrans(CUSTOM_ALPHABET, STANDARD_ALPHABET)))
            return data.decode('utf-8').strip('\x00')

        for video_id in traverse_obj(viously_players, (..., {extract_attributes}, 'id')):
            data = self._download_json(
                f'https://www.viously.com/export/json/{video_id}', video_id, transform_source=custom_decode)
            yield {
                'id': video_id,
                'formats': self._extract_m3u8_formats(f'https://www.viously.com/video/hls/{video_id}/index.m3u8', video_id),
                **traverse_obj(data, ('video', {
                    'title': 'title',
                    'description': 'description',
                    'duration': ('duration', {int_or_none}),
                    'timestamp': ('iso_date', {parse_iso8601}),
                    'categories': ('category', {lambda x: [x['name']]}),
                })),
            }
```
Tests will need to be updated.